students/rimlee_talukdar/lesson02/assignment/src/charges_calc.py

```diff
@@ -0,0 +1,168 @@
+"""
+Returns total price paid for individual rentals
+"""
+import argparse
+import json
+import datetime
+import math
+import logging
+
+
+def parse_cmd_arguments():
+    """
+    creates an argument parser for the program
+    """
+    parser = argparse.ArgumentParser(description='Process some integers.')
+    parser.add_argument('-i', '--input', help='input JSON file', required=True)
+    parser.add_argument('-o', '--output', help='ouput JSON file', required=True)
+    parser.add_argument('-d', '--debug', help='[3: ERROR, 2: WARN, 1: DEBUG, 0: NONE (default)]',
+                        required=False, type=int, default=0)
+
+    return parser.parse_args()
+
+
+def load_rentals_file(filename):
+    """
+    Opens JSON file and returns data,
+    logs a critical error if there is an error
+    """
+    try:
+        with open(filename) as file:
+            return json.load(file)
+    except FileNotFoundError as exception:
+        # Log exception if unable to open file
+        logging.critical(f'Unable to open file {filename} due to:\n\t{exception}')
+        exit(1)
+
+
+def calculate_additional_fields(cal_data):
+    """
+    Calculate the following additional field from the input data
+        1. total_days as total number of rental days
+        2. total_price as total price of the rental
+        3. sqrt_total_price as square root of the rental
+        4. unit_cost as in unit cost of the rental
+    :param cal_data: the input data
+    :return: updated data
+    """
+    for key, value in cal_data.items():
+        logging.debug('*' * 10 + f' Computing for {key} ' + '*' * 10)
+        # Update total days
+        try:
+            rental_start = datetime.datetime.strptime(value['rental_start'], '%m/%d/%y')
+            logging.debug(f'Rental start time {rental_start}')
+            rental_end = datetime.datetime.strptime(value['rental_end'], '%m/%d/%y')
+            logging.debug(f'Rental end time {rental_end}')
+
+            # If start data is after end date flip the entry
+            total_days = (rental_end - rental_start).days
+            if total_days < 0:
+                logging.debug(f'Start date after end date, hence flipping dates')
+                total_days = -total_days
+            value['total_days'] = total_days
+            logging.debug(f'Rental total days {total_days}')
+
+            value['total_price'] = total_days * value['price_per_day']
+            logging.debug(f'Rental total price {value["total_days"]}')
+
+            value['sqrt_total_price'] = math.sqrt(value['total_price'])
+            logging.debug(f'Rental square root of total price {value["total_days"]}')
+
+            value['unit_cost'] = value['total_price'] / value['units_rented']
+            logging.debug(f'Rental unit cost {value["rental_start"]}')
+
+        except Exception as exception:
```

Comment:
Be careful about logging general exceptions. You will want to specify what the possible issues would be so you know you are handling them appropriately